servicetest/filesystem/test_filesystem_strace.py

```diff
@@ -0,0 +1,140 @@
+    return CURRENT_DIR + "/test-strace.log"
+
+def agent_exec_prefix():
+    return "strace"
+
+DATA = {
+    Container.CONFIG: '[{"enableWriteBuffer": false}]',
+    Container.BIND_MOUNT_DIR: "/gateways/app",
+    Container.HOST_PATH: CURRENT_DIR,
+    Container.READONLY: False
+}
+
+def get_logblock(file, startline, stopline):
+    with open(file, "r") as input:
+        buf = ""
+        for line in input:
+            if line.strip() == startline:
+                break
+
+        for line in input:
+            if line.strip() == stopline:
+                break
+            buf += line
+
+        return buf
+
+
+def get_strace_created_files(buf):
+    files = []
+
+    for line in StringIO.StringIO(buf):
+        print line
+        if re.search('^open(.*)O_CREAT', line):
+            p = re.compile('^open\(\"(.*)\",')
+            files.append(p.match(line).group(1))
+        if re.search('^mkdir', line):
+            p = re.compile('^mkdir\(\"(.*)\",')
+            files.append(p.match(line).group(1))
+
+    return files
+
+@pytest.mark.usefixtures("create_testoutput_dir", "dbus_launch", "agent", "assert_no_proxy")
+class TestFileSystemStrace(object):
+    """ This suite should do some basic testing of the Filesystem within the
+        containers and that the whole chain is working properly.
+    """
+
+    @pytest.mark.xfail("platform.release() <= \"3.18.0\"")
+    def test_bindmount_sockets_strace(self):
+        """ Test that sockets are bindmountable as expected.
+        """
+        startword = "======STARTSTARTSTART====="
+        endword = "======ENDENDEND====="
+
+        with open(logfile_path(), mode="a") as f:
+            f.write("\n" + startword + "\n")
+
+        absolute_test_file = os.path.join("/tmp/", TESTFILE)
+
+        if not os.path.exists(absolute_test_file):
+            server = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+            server.bind(absolute_test_file)
+
+        ca = Container()
+        DATA[Container.CONFIG] = '[{"enableWriteBuffer": true}]'
+
+        try:
+            ca.start(DATA)
+            ca.bindmount(absolute_test_file, "/gateways/testfile", True)
+            # Give the command time to run inside the container
+            time.sleep(0.5)
+            assert os.path.exists(absolute_test_file) is True
+            ca.launch_command('{}/fileapp.py check {}'
+                              .format(ca.get_bind_dir(), "/gateways/testfile"))
+            time.sleep(0.5)
+            assert os.path.exists(absolute_test_file) is True
+        finally:
+            ca.terminate()
+
+        server.close()
+
+        if os.path.exists(absolute_test_file):
+            os.remove(absolute_test_file)
+
+        time.sleep(0.5)
+        with open(logfile_path(), mode="a") as f:
+            f.write("\n" + endword + "\n")
+
+        buf = get_logblock(logfile_path(), startword, endword)
+
+        files = get_strace_created_files(buf)
+
+
+        with open("gobbles.txt", "w") as w:
```

Comment:
I don't like these sort of names. I sometimes find lingering test files, log files, output etc. that are very difficult to understand where they come from. Something that supports the understanding that this is from this test would be better.